client/lib/plans/index.js

```diff
@@ -220,19 +220,30 @@ export function plansLink( url, site, intervalType ) {
 
 export function applyTestFiltersToPlansList( planName ) {
 	const filteredPlanConstantObj = { ...plansList[ planName ] };
-	let filteredPlanFeaturesConstantObj = pick( featuresList, plansList[ planName ].getFeatures() );
+	let filteredPlanFeaturesConstantList = plansList[ planName ].getFeatures();
 
 	const removeDisabledFeatures = () => {
 		if ( ! isWordadsInstantActivationEnabled() ) {
-			filteredPlanFeaturesConstantObj = pickBy( filteredPlanFeaturesConstantObj,
-				( value, key ) => key !== FEATURE_WORDADS_INSTANT
+			filteredPlanFeaturesConstantList = filter( filteredPlanFeaturesConstantList,
+				( value ) => value !== FEATURE_WORDADS_INSTANT
 			);
 		}
+
 		if ( ! isGoogleVouchersEnabled() ) {
-			filteredPlanFeaturesConstantObj = pickBy( filteredPlanFeaturesConstantObj,
-				( value, key ) => key !== FEATURE_GOOGLE_AD_CREDITS
+			filteredPlanFeaturesConstantList = filter( filteredPlanFeaturesConstantList,
+				( value ) => value !== FEATURE_GOOGLE_AD_VOUCHERS_100 &&
+					value !== FEATURE_GOOGLE_WORDADS_AD_VOUCHERS_200
 			);
 		}
+
+		if ( ! isWordpressAdCreditsEnabled() ) {
+			const wordpressAdCreditsFeatureIndex = filteredPlanFeaturesConstantList.
+				indexOf( FEATURE_GOOGLE_WORDADS_AD_VOUCHERS_200 );
+
+			if ( wordpressAdCreditsFeatureIndex !== -1 ) {
+				filteredPlanFeaturesConstantList[ wordpressAdCreditsFeatureIndex ] = FEATURE_GOOGLE_AD_VOUCHERS_100;
```

Comment:
What is meant to happen here? I'm not really clear what we're doing, but it doesn't appear that we are removing a disabled feature. I don't believe we need to check anything about WordPressAdCredits in this function at all, since it's considered part of the same feature as google vouchers (and is not supposed to existing without google vouchers anyway).